uniffi_bindgen/src/bindings/swift/templates/ErrorTemplate.swift

```diff
@@ -9,6 +9,47 @@ protocol RustError: LocalizedError {
     static func fromConsuming(_ rustError: NativeRustError) throws -> Self?
 }
 
+// An error type for FFI errors. These errors occur at the UniFFI level, not
+// the library level.
+enum InternalError: RustError {
+    case bufferOverflow
+    case incompleteData
+    case unexpectedOptionalTag
+    case unexpectedEnumCase
+    case emptyResult
+    case panic(message: String)
+    case invalidHandle(message: String)
+    case unknownError(code: Int32)
```

Comment:
Possibly contrary opinion: I'm not convinced it's valuable to define an elaborate series of cases for each individual type of internal error that might be thrown. If anyone ever finds themselves matching on these different cases and doing anything other than freaking out and filing a bug against uniffi, something has gone terribly wrong 😅.

In that spirit, the most important thing is that these errors surface with a useful human-readable message that helps us find and fix the bug. If enumerating all the cases helps us achieve that, awesome. But if they're just adding layers of indirection to a fixed string value, then I worry they might set incorrect expectations about how these internal errors should be handled.

I'm not going to r- for that though, just my 2c.

Reply:
Hmm, thinking some more about this, I'm coming around to what you and Tarik suggested—assert for any kind of internal error instead of bubbling it up as an exception. In real-world code, `[Throws]` is for defining error cases that are meaningfully recoverable—a network error or duplicate experiment name, for example. A Rust-side panic _might_ be, but...how do you fix that? If the app were fully in Rust, it would crash. The only reason we catch panics is to avoid stack unwinding into the foreign language, not as a graceful error recovery mechanism.